docker/importHistograms.py

```diff
@@ -0,0 +1,136 @@
+import hashlib
+import json
+import sys
+import urllib.request
+
+from google.api_core.exceptions import DeadlineExceeded
+from google.cloud import bigquery, firestore
+
+
+# The number of historical versions to import for each channel.
+NUM_VERSIONS = 10
+# The channels to import data for.
+CHANNELS = ("release", "beta", "nightly")
+
+
+PRODUCT_DETAILS_URL = (
+    "https://product-details.mozilla.org/1.0/firefox_versions.json"
+)
+BQ_TABLE = "moz-fx-data-derived-datasets.telemetry.client_probe_counts_v1"
+BIGQUERY = bigquery.Client()
+FIRESTORE = firestore.Client()
+BATCH_RETRIES = []
+
+
+def commit_batch(batch):
+    try:
+        batch.commit()
+        print(".", end="", flush=True)
+    except DeadlineExceeded:
+        BATCH_RETRIES.append(batch)
+        print("F", end="", flush=True)
+    except:  # noqa
+        print("Unexpected error:", sys.exc_info()[0])
+
+    return FIRESTORE.batch()
+
+
+def import_collection(channel, version):
+
+    query = """
+        SELECT *
+        FROM `{table}`
+        WHERE
+            channel = @channel
+            AND app_version = @app_version
+            AND os IS NOT NULL
+            AND app_build_id IS NOT NULL
+    """.format(
+        table=BQ_TABLE
+    )
+    params = [
+        bigquery.ScalarQueryParameter("channel", "STRING", channel),
+        bigquery.ScalarQueryParameter("app_version", "STRING", version),
+    ]
+    config = bigquery.QueryJobConfig()
+    config.query_parameters = params
+    job = BIGQUERY.query(query, job_config=config)
+
+    print(
+        "Inserting batched documents for {channel}-{version} ...".format(
+            **vars()
+        )
+    )
+
+    i = None
+    batch = FIRESTORE.batch()
+
+    for i, row in enumerate(job):
+
+        if i % 500 == 0 and i > 0:
+            batch = commit_batch(batch)
+
+        doc = {
+            "metric": row.metric,
+            "build_id": row.app_build_id,
+            "os": row.os,
+            "agg_type": row.agg_type,
+            "aggregates": row.aggregates["key_value"],
```

Comment:
Just a heads up that `key_value` got removed from `main_summary` as an intermediate. So I tried to follow suit and remove it from `client_probe_counts_v2`. So I think once you use that one, you might just be able to use `row.aggregates` directly instead of `row.aggregates["key_value"]`.

Reply:
Ah, good catch. I didn't even think that I was still using the `v1` table.